src/libcore/cmp.rs

```diff
@@ -512,16 +571,56 @@ impl PartialOrd for Ordering {
     }
 }
 
-/// Trait for values that can be compared for a sort-order.
+/// Trait for values that can be partially ordered.
+///
+/// This traits defines the operators `<`, `>`, `<=`, and `>=` which compare two values and return `bool`, 
+/// where `<` and `>` form [_strict_ partial ordering relations][wiki_porder].
+///
+/// The `PartialOrd` trait  has fewer requirements of the ordering relation than Ord. In particular, `partial_cmp` 
+/// can return `None` if two elements cannot be compared. That means that `PartialOrd` can be implemented 
+/// for types with incomparable values like floating-point numbers (`NaN`) and for ordering relations like 
+/// subset relations.
+///
+/// It extends the partial equivalence relation provided by `PartialEq` (`==`) with `partial_cmp(a, b) -> Option<Ordering>`, 
+/// which is a [trichotomy](https://en.wikipedia.org/wiki/Trichotomy_(mathematics)) of the ordering relation when 
+/// its result is `Some`:
+///
+/// * `a < b` if and only if `partial_cmp(a, b) == Some(Less)`
+/// * `a > b` if and only if `partial_cmp(a, b) == Some(Greater)`
+/// * `a == b` if and only if `partial_cmp(a, b) == Some(Equal)`
 ///
-/// The comparison must satisfy, for all `a`, `b` and `c`:
+/// and the absence of an ordering between `a` and `b` if and only if `partial_cmp(a, b) == None`. 
+/// Furthermore, this trait defines `<=` as `a < b || a == b` and `>=` as `a > b || a == b`.
 ///
-/// - antisymmetry: if `a < b` then `!(a > b)`, as well as `a > b` implying `!(a < b)`; and
-/// - transitivity: `a < b` and `b < c` implies `a < c`. The same must hold for both `==` and `>`.
+/// The comparisons must satisfy, for all `a`, `b`, and `c`:
 ///
-/// Note that these requirements mean that the trait itself must be implemented symmetrically and
-/// transitively: if `T: PartialOrd<U>` and `U: PartialOrd<V>` then `U: PartialOrd<T>` and `T:
-/// PartialOrd<V>`.
+/// * _transitivity_: if `a < b` and `b < c` then `a < c`
+/// * _duality_: if `a < b` then `b > a`
+///
+/// The `lt` (`<`), `le` (`<=`), `gt` (`>`), and `ge` (`>=`) methods are implemented in terms of `partial_cmp` 
+/// according to these rules. The default implementations can be overridden for performance reasons, but manual 
+/// implementations must satisfy the rules above.
+///
+/// From these rules it follows that `PartialOrd` must be implemented symmetrically and transitively.
+/// That is, for two distinct types `T` and `U`, `T: PartialOrd<U>` requires the following trait 
+/// implementations to exist and satisfy these rules: `T: PartialOrd<T>`, `U: PartialOrd<T>`, 
+/// and `U: PartialOrd<U>`. If for a third distinct type `V` `T: PartialOrd<V>` is provided, then
```

Comment:
Mhhh I don't quite get why `T: PartialOrd<T>` and `U: PartialOrd<U>` are required from the rules above. I am probably just missing something obvious...

Reply:
This is required by transitivity (if `a < b` and `b < c` then `a < c`).

Suppose that `a: T`, `b: U`, and `c: T`., and that `a < b` and `b < c` are both true. For `a < c` to also be true it would need to be well-typed, and that requires `T: PartialOrd<T>`.

Reply:
Ah right, that makes sense. Thanks :)